viper.py

```diff
@@ -724,6 +750,38 @@ def fit_chunk(o, chunk, obsname, targ=None, tpltarg=None):
     print(file=parunit)
     #vpr.plot_rvo(rv, e_rv)
 
+
+if createtpl:
+    wtpl_new = {}
+    tpl_new = {}
+    for o in orders:
+        gplot.reset()
+        gplot.key("title 'order: %s' noenhance" % (o))
+        wt = np.array(list(tpl_all[o,0].values())) 	# wavelength
+        ft = np.array(list(tpl_all[o,1].values()))	# data
+        mt = np.array(list(tpl_all[o,2].values()))	# weighting
+        mt[np.isnan(ft)] = np.nan
+        mt[ft<0] = np.nan
+
+        mt[ft>1.15] = np.nanmin(mt)/10.
+        tpl_new[o] = np.nansum(ft*mt,axis=0) / np.nansum(mt,axis=0)
+        wtpl_new[o] = np.nanmean(wt,axis=0)
+
+        # flag outlier points and spectra
+        for nn in range(0,len(ft),1):
+            mt[nn][np.abs(ft[nn]-tpl_new[o])>0.1] = np.nan
+
+        tpl_new[o] = np.nansum(ft*mt,axis=0) / np.nansum(mt,axis=0)
+
+        gplot(wtpl_new[o], tpl_new[o] - 1 ,'w l lc 7 t "combined tpl"')
+        for n in range(0,len(ft),1):
+            gplot+(wtpl_new[o], ft[n]/np.nanmean(ft[n]),'w l t "%s"' % (os.path.split(obsnames[n])[1])) 
```

Comment:
Please post a plot illustrating the template creation.

Reply:
use again `os.path.basename`.

Reply:
Here is an example for the plot. While the combined template is plotted in the bottom, you find the single corrected spectra plotted above each other in the top.

![grafik](https://user-images.githubusercontent.com/77435060/201129542-286ded54-2528-44ae-8f28-094f1baca478.png)

Reply:
Thx. Very helpful.